src/utils/salesEvidenceFilter.ts

```diff
@@ -0,0 +1,47 @@
+import { SalesEvidenceFilterParams, SaleTypeEnum, UnprocessedResultsFromCRM, MinMaxNumberType, MinMaxDateType } from '../types'
+
+function numberFilter (filterType: string, filterValues: MinMaxNumberType, property: UnprocessedResultsFromCRM) {
+    return typeof property[filterType] === 'number' && property[filterType] >= filterValues.min && property[filterType] <= filterValues.max
+}
+
+function dateFilter (dateSold: MinMaxDateType, property: UnprocessedResultsFromCRM): boolean {
+    if (typeof dateSold.min !== 'undefined' && typeof dateSold.max !== 'undefined') {
+        const minDate = dateSold.min.toISOString().split('T')[0]
+        const maxDate = dateSold.max.toISOString().split('T')[0]
+        return !property.Sale_Date && property.Sale_Date >= minDate && property.Sale_Date <= maxDate
+    }
+    return false
+}
+
+function saleTypeFilter (saleTypes: SaleTypeEnum[], property: UnprocessedResultsFromCRM): boolean {
+    return saleTypes.some((saleType: SaleTypeEnum) => {
+        return property.Sale_Type.includes(saleType)
+    })
+}
+
+export default function salesEvidenceFilter (filterParameters: SalesEvidenceFilterParams, property: UnprocessedResultsFromCRM): boolean {
+    const {
+        landArea,
+        buildArea,
+        salePrice,
+        saleType,
+        dateSold
+    } = filterParameters
+    const BLANK_FILTER_VALUE = -1
+    const isLandAreaFilterInUse = landArea.min === BLANK_FILTER_VALUE && landArea.max === BLANK_FILTER_VALUE
+    const isInLandAreaRange = !isLandAreaFilterInUse && numberFilter('Land_Area_sqm', landArea, property)
```

Comment:
Inverse logic. Should be isLandAreaFilterInUse not ! isLandAreaFilterInUse

Reply:
I think the variable is not correctly named. When it's `true` it means the filter is not in use b/c it is equal to `-1` which is the default value of the field. It is only when this is equal to `false` when the filter is in use which is why I have bang operator being used here.